### PR TITLE
Fix fragile cache name property pattern for _xpehh_gwss_cache_name and _ihs_gwss_cache_name

### LINKED ISSUE #1151: Error accessing roh_hmm
An error is raised accessing the ag3.roh_hmm function as below.

NotImplementedError Traceback (most recent call last)
/tmp/ipykernel_264/1691514004.py in <cell line: 0>()
----> 1 roh_plot_df=get_roh_counts(region=['3L'], sample_query='sample_set == "1273-VO-ZM-MULEBA-VMF00176" and taxon == "gambiae"', site_mask='gamb_colu', window_size=10_000)

3 frames
/usr/local/lib/python3.12/dist-packages/malariagen_data/anopheles.py in _roh_hmm_cache_name(self)
564 @AbstractMethod
565 def _roh_hmm_cache_name(self):
--> 566 raise NotImplementedError("Must override _roh_hmm_cache_name")
567
568 @_check_types

NotImplementedError: Must override _roh_hmm_cache_name

It seems this error stems from trying to access the internal function _roh_hmm_cache_name. Possibly this does not return a doc string as expected when setting up the results cache.

### PR DESCRIPTION
**Pull request by kunal-10-cloud**
## Description

This PR fixes a fragile pattern in `_xpehh_gwss_cache_name` and `_ihs_gwss_cache_name` that could cause `NotImplementedError` crashes under MRO edge cases or version skew. The fix applies the same safe resolver pattern established in #1193 for `_roh_hmm_cache_name`.

Fixes: #1198

### anopheles.py
- Removed `from abc import abstractmethod` (no longer needed)
- Replaced abstract `_xpehh_gwss_cache_name` property with `_get_xpehh_gwss_cache_name()` resolver method
- Replaced abstract `_ihs_gwss_cache_name` property with `_get_ihs_gwss_cache_name()` resolver method
- Updated `ihs_gwss()` call site (line ~742) to use `_get_ihs_gwss_cache_name()`
- Updated `xpehh_gwss()` call site (line ~1263) to use `_get_xpehh_gwss_cache_name()`

### Species Classes (af1.py, adar1.py, ag3.py, adir1.py, amin1.py)
- Removed `_xpehh_gwss_cache_name` class attribute override
- Removed `_ihs_gwss_cache_name` class attribute override
- Cache name constants remain intact for reference

## Type of Change

- [x] Bug fix (fixes latent crash pattern)
- [ ] New feature
- [ ] Breaking change
- [ ] Documentation update

## Testing

[X] Python syntax validation passed
[X] Ruff linting passed
[X] Ruff formatting passed
[X] Mypy type checking passed (no issues found)
[X] Pre-commit hooks passed

## Related Issues

Fixes #1151 pattern for two additional cache name properties. Implements the same solution pattern established in #1193 for `_roh_hmm_cache_name`.

## Checklist

- [X] Code follows project style guidelines (ruff, mypy)
- [X] Pre-commit hooks pass
- [X] Changes are properly tested
- [X] Documentation reflects changes
- [X] No new warnings introduced

### CODE REVIEW

File: tests/anoph/test_cache_name_resolvers.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
+"""Test cache name resolver methods for MRO edge case handling."""
```
[the remark this answers is not in the document]
Comment (kunal-10-cloud) [reply tests/anoph/test_cache_name_resolvers.py]: you are right, I shouldn't have made a different test file 
The resolvers are tested implicitly through existing integration tests 
(test_xpehh_gwss and test_ihs_gwss in test_ag3.py) that exercise these methods. 
If the resolver breaks, those tests fail.
